pkg/loki/modules.go

```diff
@@ -1426,11 +1428,13 @@ func (t *Loki) initQuerySchedulerRing() (_ services.Service, err error) {
 	// Set some config sections from other config sections in the config struct
 	t.Cfg.QueryScheduler.SchedulerRing.ListenPort = t.Cfg.Server.GRPCListenPort
 
-	managerMode := scheduler.RingManagerModeReader
+	managerMode := lokiring.ClientMode
 	if t.Cfg.isModuleEnabled(QueryScheduler) || t.Cfg.isModuleEnabled(Backend) || t.Cfg.isModuleEnabled(All) || (t.Cfg.LegacyReadTarget && t.Cfg.isModuleEnabled(Read)) {
-		managerMode = scheduler.RingManagerModeMember
+		managerMode = lokiring.ServerMode
 	}
-	rm, err := scheduler.NewRingManager(managerMode, t.Cfg.QueryScheduler, util_log.Logger, prometheus.DefaultRegisterer)
+	rf := 2     // ringReplicationFactor should be 2 because we want 2 schedulers.
+	tokens := 1 // we only need to insert 1 token to be used for leader election purposes.
+	rm, err := lokiring.NewRingManager("scheduler", managerMode, t.Cfg.QueryScheduler.SchedulerRing, rf, tokens, util_log.Logger, prometheus.DefaultRegisterer)
```

Comment:
Let's put all the `ringKey`s into a const declaration like

```
const (
  schedulerRingKey="scheduler"
  ..etc
)
```

Reply:
See https://github.com/grafana/loki/pull/10931/commits/a276cb68bef1b9cb70d959d366dcfc65632ead8d